rotkehlchen/accounting/rules.py

```diff
@@ -0,0 +1,170 @@
+from typing import TYPE_CHECKING, Optional
+
+from rotkehlchen.accounting.structures.base import HistoryBaseEntry
+from rotkehlchen.accounting.structures.evm_event import EvmEvent
+from rotkehlchen.accounting.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.chain.evm.accounting.structures import BaseEventSettings, TxAccountingTreatment
+from rotkehlchen.chain.evm.decoding.constants import CPT_GAS
+from rotkehlchen.db.constants import NO_ACCOUNTING_COUNTERPARTY
+
+if TYPE_CHECKING:
+    from rotkehlchen.accounting.pot import AccountingPot
+    from rotkehlchen.chain.evm.accounting.aggregator import EVMAccountingAggregators
+    from rotkehlchen.db.dbhandler import DBHandler
+
+
+def get_accounting_rule_identifier(
+        event_type: HistoryEventType,
+        event_subtype: HistoryEventSubType,
+        counterparty: Optional[str] = None,
+) -> str:
+    key = f'{event_type.serialize()}_{event_subtype.serialize()}'
```

Comment:
Just as we discussed in our call just concatenate them fren. There is no reason to keep the `_`. The reason for the `__` was to maybe at some point get the type/subtype/counterparty back.

But since that's not needed let's make it an one way function to save space also. So not only contatenate but also **hash**. This will not make it human readable anymore but does it matter?

If not, and I think it does not, then we will end up having much smaller keys.